DragaliaAPI/Features/Missions/MissionInitialProgressionService.cs

```diff
@@ -14,81 +15,104 @@
+                x => x.MissionType == mission.Type && x.MissionId == mission.Id
             );
 
         if (progressionInfo == null)
             return;
 
-        MissionProgressionRequirement requirement = progressionInfo.Requirements.First(
-            x => x.Missions.Contains(info)
-        );
-
         int amountToComplete = missionInfo.CompleteValue;
-        int currentAmount = progressionInfo.Type switch
+        int currentAmount = progressionInfo.CompleteType switch
         {
-            MissionProgressType.FortPlantBuilt
-                => await this.fortRepository.Builds.CountAsync(
-                    x => x.PlantId == (FortPlants)requirement.Parameter!
+            MissionCompleteType.FortPlantLevelUp
+                => await fortMissionProgressionService.GetMaxFortPlantLevel(
+                    (FortPlants)progressionInfo.Parameter!
                 ),
-            MissionProgressType.FortPlantUpgraded
-                => await this.fortRepository.Builds
-                    .Where(x => x.PlantId == (FortPlants)requirement.Parameter!)
-                    .Select(x => x.Level)
-                    .FirstOrDefaultAsync(),
-            MissionProgressType.FortLevelup
-                => await this.fortRepository.Builds
-                    .Where(x => x.PlantId != FortPlants.TheHalidom)
-                    .SumAsync(
-                        x => x.BuildEndDate == DateTimeOffset.UnixEpoch ? x.Level : x.Level - 1
-                    ),
-            MissionProgressType.QuestCleared => await GetQuestClearedCount(requirement),
-            MissionProgressType.CharacterBuildup => await GetCharacterBuildupCount(requirement),
-            MissionProgressType.WyrmprintAugmentBuildup
-                => await GetWyrmprintBuildupCount(requirement),
-            MissionProgressType.WeaponEarned => await GetWeaponEarnedCount(requirement),
-            MissionProgressType.WeaponRefined => await GetWeaponRefinedCount(requirement),
-            MissionProgressType.VoidBattleCleared => await GetVoidBattleClearedCount(requirement),
-            MissionProgressType.ItemSummon => 0, // TODO: As daily quests also use this, should we make it actually count the item summons?
+            MissionCompleteType.FortPlantBuilt
+            or MissionCompleteType.FortPlantPlaced
+                => await fortMissionProgressionService.GetFortPlantCount(
+                    (FortPlants)progressionInfo.Parameter!
+                ),
+            MissionCompleteType.FortLevelUp
+                => await fortMissionProgressionService.GetTotalFortLevel(),
+            MissionCompleteType.QuestCleared => await GetQuestClearedCount(progressionInfo),
+            MissionCompleteType.QuestStoryCleared
+                => await storyRepository.QuestStories.AnyAsync(
+                    x => x.StoryId == progressionInfo.Parameter && x.State == StoryState.Read
+                )
+                    ? 1
+                    : 0,
```

Comment:
can you use StoryRepository.HasReadQuestSTory?

Reply:
yes, changed